saleor/webhook/transport/shipping.py

```diff
@@ -1,106 +1,43 @@
 
 logger = logging.getLogger(__name__)
 
 
-def to_shipping_app_id(app: App, shipping_method_id: str) -> str:
-    app_identifier = app.identifier or app.id
-    return base64.b64encode(
-        str.encode(f"{APP_ID_PREFIX}:{app_identifier}:{shipping_method_id}")
-    ).decode("utf-8")
-
-
-def convert_to_app_id_with_identifier(shipping_app_id: str) -> None | str:
-    """Prepare the shipping_app_id in format `app:<app-identifier>/method_id>`.
-
-    The format of shipping_app_id has been changes so we need to support both of them.
-    This method is preparing the new shipping_app_id format based on assumptions
-    that right now the old one is used which is `app:<app-pk>:method_id>`
-    """
-    decoded_id = base64.b64decode(shipping_app_id).decode()
-    splitted_id = decoded_id.split(":")
-    if len(splitted_id) != 3:
-        return None
-    try:
-        app_id = int(splitted_id[1])
-    except (TypeError, ValueError):
-        return None
-    app = App.objects.filter(id=app_id).first()
-    if app is None:
-        return None
-    return to_shipping_app_id(app, splitted_id[2])
-
-
-def method_metadata_is_valid(metadata) -> bool:
-    if not isinstance(metadata, dict):
-        return False
-    for key, value in metadata.items():
-        if not isinstance(key, str) or not isinstance(value, str) or not key.strip():
-            return False
-    return True
-
-
 def parse_list_shipping_methods_response(
     response_data: Any, app: "App"
 ) -> list["ShippingMethodData"]:
-    shipping_methods = []
-    for shipping_method_data in response_data:
-        if not validate_shipping_method_data(shipping_method_data):
-            continue
-        method_id = shipping_method_data.get("id")
-        method_name = shipping_method_data.get("name")
-        method_amount = shipping_method_data.get("amount")
-        method_currency = shipping_method_data.get("currency")
-        method_maximum_delivery_days = shipping_method_data.get("maximum_delivery_days")
-        method_minimum_delivery_days = shipping_method_data.get("minimum_delivery_days")
-        method_description = shipping_method_data.get("description")
-        method_metadata = shipping_method_data.get("metadata") or {}
-        if method_metadata:
-            method_metadata = (
-                method_metadata if method_metadata_is_valid(method_metadata) else {}
-            )
-
-        shipping_methods.append(
-            ShippingMethodData(
-                id=to_shipping_app_id(app, method_id),
-                name=method_name,
-                price=Money(method_amount, method_currency),
-                maximum_delivery_days=method_maximum_delivery_days,
-                minimum_delivery_days=method_minimum_delivery_days,
-                description=method_description,
-                metadata=method_metadata,
-            )
+    try:
+        list_shipping_method_model = ListShippingMethodsSchema.model_validate(
```

Comment:
Is there any reason why you use `model_validate` explicitly? 
Personally I would just call `ListShippingMethodsSchema(response_data)` as this would be more friendly from me (but maybe I am missing something 😓 )

Reply:
Using `model_validate()` is the recommended way, as it allows to pass additional validation parameters like `by_alias` or `context`. We should consider using `model_validate_json` and passing raw JSON directly to it as it is [recommended performance-wise](https://docs.pydantic.dev/2.11/concepts/performance/#in-general-use-model_validate_json-not-model_validatejsonloads)